build/dotnet/dotnet-install.sh

```diff
@@ -549,10 +549,6 @@ download() {
     local out_path=${2:-}
 
     local failed=false
-    # Restart the request up to N times if it fails
```

Comment:
I think we should try reverting these changes and see if it causes issues (e.g. on Travis).  If so, we should submit a PR to https://github.com/dotnet/cli rather than change our copy.